### PR TITLE
[gfx94x] Add new target ID changes to static CK kernels

### PR DESCRIPTION
**Pull request by muralinr**
Adding new target ID changes

### CODE REVIEW

File: src/composable_kernel/composable_kernel/include/utility/config.hpp [hunk 1]
```diff
@@ -14,8 +14,9 @@
 // should enable one and only one GPU target
 #if !(defined(CK_AMD_GPU_GFX803) || defined(CK_AMD_GPU_GFX900) || defined(CK_AMD_GPU_GFX906) ||    \
       defined(CK_AMD_GPU_GFX940) || defined(CK_AMD_GPU_GFX908) || defined(CK_AMD_GPU_GFX90A) ||    \
-      defined(CK_AMD_GPU_GFX1030) || defined(CK_AMD_GPU_GFX1031) || defined(CK_AMD_GPU_GFX1100) || \
-      defined(CK_AMD_GPU_GFX1101) || defined(CK_AMD_GPU_GFX1102))
+      defined(CK_AMD_GPU_GFX941) || defined(CK_AMD_GPU_GFX942) || defined(CK_AMD_GPU_GFX1030) ||   \
```
Comment (atamazov) [inline src/composable_kernel/composable_kernel/include/utility/config.hpp:17]: [Recommendation] @muralinr In similar cases, please add new items to the end -- to make diffs smaller and reviewer's job easier.

File: src/solver/batchnorm/backward_spatial_single.cpp [hunk 2]
```diff
@@ -230,7 +230,8 @@ BnBwdTrainingSpatialSingle::GetSolution(const ExecutionContext& context,
 #if WORKAROUND_ISSUE_1146
              && (handle.GetDeviceName() != "gfx90a")
 #endif
-             && (handle.GetDeviceName() != "gfx940"))) &&
+             && (handle.GetDeviceName() != "gfx940") && (handle.GetDeviceName() != "gfx941") &&
+             (handle.GetDeviceName() != "gfx942"))) &&
```
Comment (atamazov) [inline src/solver/batchnorm/backward_spatial_single.cpp]: [Recommandation]
```suggestion
             && !StartsWith(handle.GetDeviceName(), "gfx94")) &&
```

**Review by atamazov — APPROVED** [review]
I do not see anything bad here. However, I didn't check if the new targets were added in _all the necessary_ places.

File: src/composable_kernel/composable_kernel/include/utility/config.hpp (hunk 1 above)
Comment (muralinr) [reply src/composable_kernel/composable_kernel/include/utility/config.hpp:17]: Sure, will add going forward.

**Review by muralinr — COMMENTED** [review]
Approved one of your recommended changes

File: src/composable_kernel/composable_kernel/include/utility/config.hpp (hunk 1 above)
Comment (JehandadKhan) [reply src/composable_kernel/composable_kernel/include/utility/config.hpp:17]: Since static CK has been deprecated for the new targets, do we need this ?

**Review by JehandadKhan — DISMISSED** [review]
These changes appear to affect static CK only, do we need this changes to begin with ? Since static CK has been deprecated for newer ASICs.

**Comment by muralinr** [comment]
> These changes appear to affect static CK only, do we need this changes to begin with ? Since static CK has been deprecated for newer ASICs.

Yes, we need to these static CK changes to pass some of cmake tests.

File: src/composable_kernel/composable_kernel/include/utility/config.hpp (hunk 1 above)
Comment (muralinr) [reply src/composable_kernel/composable_kernel/include/utility/config.hpp:17]: Yes, we need to these static CK changes to pass some tests in test folder.

**Review by junliume — APPROVED** [review]

File: src/composable_kernel/composable_kernel/include/utility/config.hpp (hunk 1 above)
Comment (atamazov) [reply src/composable_kernel/composable_kernel/include/utility/config.hpp:17]: @JehandadKhan Yeah, while we maintain possibility to enable deprecated kernels for the new targets.

**Comment by JehandadKhan** [comment]
> Yes, we need to these static CK changes to pass some of cmake tests.

@muralinr  Perhaps those tests should be properly updated so they dont run Static CK.

**Comment by muralinr** [comment]
> > Yes, we need to these static CK changes to pass some of cmake tests.
> 
> @muralinr Perhaps those tests should be properly updated so they dont run Static CK.

Yes @JehandadKhan , that would be long term goal.  We need this target changes this week for gfx941 target bring up.

**Pull request merged**